app/src/main/java/com/github/se/icebreakrr/ui/sections/UnblockProfile.kt

```diff
@@ -0,0 +1,248 @@
+          dismissButton = {
+              TextButton(onClick = { profileToUnblock = null }) {
+                  Text("No")
+              }
+          }
+      )
+  }
+
+  // Initial check and start of periodic update every 10 seconds
+  LaunchedEffect(isConnected.value) {
+    if (!isTestMode && !isNetworkAvailable()) {
+      profilesViewModel.updateIsConnected(false)
+    } else {
+      while (true) {
+        // Call the profile fetch function
+        profilesViewModel.getBlockedUsers()
+
+        // Wait 10 seconds before the next update
+        delay(10_000L)
+      }
+    }
+  }
+
+  Scaffold(
+      modifier = Modifier.testTag("unblockScreen"),
+      bottomBar = {
+          BottomNavigationMenu(
+              onTabSelect = { route ->
+                  if (navigationActions.currentRoute() != Route.SETTINGS && route.route == Route.SETTINGS) {
+                      navigationActions.navigateTo(Route.SETTINGS)
+                  }
+                  else {
+                      navigationActions.navigateTo(route.route)
+                  }
+              },
+              tabList = LIST_TOP_LEVEL_DESTINATIONS,
+              selectedItem = Route.UNBLOCK_PROFILE)
+      },
+      topBar = {
+          TopAppBar(
+              modifier = Modifier.testTag("unblockTopBar"),
+              title = { Text("Blocked Users",color= Color.White, modifier = Modifier.testTag("UnblockTopBarTitle")) },
+              navigationIcon = {
+                  IconButton(
+                      modifier = Modifier.testTag("goBackButton"),
+                      onClick = {
+                          navigationActions.goBack()
+                      }) {
+                      Icon(
+                          Icons.AutoMirrored.Filled.ArrowBack,
+                          contentDescription = "Back",
+                          tint = Color.White)
+                    }
+              },
+              colors = TopAppBarDefaults.topAppBarColors(containerColor = IcebreakrrBlue))
+      },
+      content = { innerPadding ->
+        PullToRefreshBox(
+            isRefreshing = isLoading.value,
+            onRefresh = {profilesViewModel.getSelfProfile(); profilesViewModel.getBlockedUsers()},
+            modifier = Modifier.padding(innerPadding)
+        ) {
+              LazyColumn(
+                  contentPadding = PaddingValues(vertical = COLUMN_VERTICAL_PADDING),
+                  verticalArrangement = Arrangement.spacedBy(COLUMN_VERTICAL_PADDING),
+                  modifier =
+                      Modifier.fillMaxSize().padding(horizontal = COLUMN_HORIZONTAL_PADDING)) {
+                    if (!isConnected.value) {
+                      item {
+                        Box(
+                            contentAlignment = Alignment.Center,
+                            modifier = Modifier.fillMaxSize().testTag("noConnectionPrompt")) {
+                              Text(
+                                  text = "No Internet Connection",
```

Comment:
Nitpick: Hardcoded string could be put in Strings.xml or there might already be one for this I think.